### ISSUE TITLE
Conditionally rendering a slot doesn't update position correctly

### ISSUE THREAD

**Issue by corysmc** [ionic-team/stencil#721]
**Stencil version:**

```
 @stencil/core@0.7.20
```

**I'm submitting a:**
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**
Using an `ion-item` and conditionally rendering text inside `slot="end"` does show the text, but it isn't placed at the end, where it should be placed.
Remove the conditional and it renders in the correct place.

**Expected behavior:**
Should be able to conditionally render a component with a named slot, and when re-rendered the content should display in the proper place.

**Steps to reproduce:**
```tsx
<ion-item>
  <h2>
    LABEL HERE
  </h2>
  {this.level == this.activePlanLevel ? (
    <h3 class="selected-plan-badge" slot="end">
      SELECTED
    </h3>
    ) : null}
</ion-item>
```

**Related code:**
This works (because the slot is already there, we're just populating it with text)
```tsx
<ion-item>
  <h2>
    LABEL HERE
  </h2>
  <h3 class="selected-plan-badge" slot="end">
  {this.level == this.activePlanLevel? 'SELECTED' : null}
  </h3>
</ion-item>
```

**Comment by corysmc** [comment]
![image](https://user-images.githubusercontent.com/6452188/38582855-5f7a9104-3cce-11e8-8537-047a4ce4a07b.png)
Broken Code (slot renders, just not in correct position) ☝️ 
Related Code (how it should look) 👇 
![image](https://user-images.githubusercontent.com/6452188/38582894-7f6b9396-3cce-11e8-9b21-e4e1dcc2d4c2.png)

**Comment by julesjanssen** [comment]
I'm pretty sure something broke between v0.7.10 and 0.7.11 that causes this. We're having an issue a lot like this. On 0.7.10 it works as expected, from 0.7.11 onwards it doesn't anymore.

**Comment by adamdbradley** [comment]
@julesjanssen would you be able to try the last version? We've since fixed many of the slot bugs.

**Comment by julesjanssen** [comment]
Yeah, it doesn't work with v0.7.24. I actually noticed the bug on that version and then tried a bunch of different versions to pinpoint where it breaks.

**Comment by corysmc** [comment]
Yep still exists in v0.7.24. Here's another example:

```tsx
<ion-list>
  {this.categoryWorkouts.length
    ? this.categoryWorkouts.map((workout: WorkoutObject) => (
        <ion-item detail href={`/workouts/${workout.slug}`}>
          <ion-avatar slot="start">
            <img src={workout.image} />
          </ion-avatar>
          <div>
            <h2>{workout.name}</h2>
            <p>{workout.description}</p>
          </div>
        </ion-item>
      ))
    : Array.from(Array(10).keys()).map(() => (
        <ion-item detail>
          <ion-avatar slot="start">
            <div class="sw-skeleton image circle" />
          </ion-avatar>
          <div>
            <h2 class="sw-skeleton text" />
            <p class="sw-skeleton text" />
          </div>
        </ion-item>
      ))}
</ion-list>;
```
On render of the skeleton item, the avatar placeholder is in the right place. But on re-render (when workouts are loaded) the avatar isn't displayed at all.

**Comment by julesjanssen** [comment]
Are you sure this fixes the issue?
Just tried to build a component with conditional slot again with v0.7.27-0 of stencil/core and the slot still ends up empty if the condition switches.

**Comment by adamdbradley** [comment]
@julesjanssen can you provide an example so we can test it?

**Comment by julesjanssen** [comment]
@adamdbradley My bad, in a simple example it works like a charm so there's probably something else in the component that causes an issue here.

**Comment by julesjanssen** [comment]
Created another example that does change the order of two named slots which seems to be a problem.
https://gist.github.com/julesjanssen/169be8004b44f0a09a9e0d2f33014b25

**Comment by adamdbradley** [comment]
Would you be able to try out `0.7.27-1`? It has some slot reorder fixes.

**Comment by adamdbradley** [comment]
@juanmiguelbesada would you please be able to provide an example which replicates this issue?

**Comment by juanmiguelbesada** [comment]
Here you go https://github.com/juanmiguelbesada/stencil-conditionally-slot

Based on @julesjanssen gist

**Comment by juanmiguelbesada** [comment]
Also note that the div is getting a "no" class each time.

``` 
<div class="no no no no no">...</div>
```

**Comment by adamdbradley** [comment]
This is the karma test I've been using to test it. Trying to figure out what it's missing: https://github.com/ionic-team/stencil/tree/master/test/karma/src/slot-reorder

**Comment by adamdbradley** [comment]
@juanmiguelbesada would you be able to test out 🏖 0.7.27-2? Also, are you available to jump on slack ? https://stencil-worldwide.herokuapp.com/

**Comment by julesjanssen** [comment]
My earlier example works great with 🏖 0.7.27-2. Thanks.

**Comment by juanmiguelbesada** [comment]
Yes! Fixed in 0.7.27-2!
